Tools/TestWebKitAPI/Tests/IPC/IPCSerialization.mm

```diff
@@ -549,8 +549,10 @@ static BOOL wkDDActionContext_isEqual(WKDDActionContext *a, SEL, WKDDActionConte
 
 static bool wkNSURLProtectionSpace_isEqual(NSURLProtectionSpace *a, SEL, NSURLProtectionSpace* b)
 {
-    if (![a.host isEqual: b.host])
-        return false;
+    if (![a.host isEqual: b.host]) {
```

Comment:
`if (a.host != b.host && ![a.host isEqual: b.host])`
Why don't we need this check for the other params like protocol below?

Reply:
host is not nullable (the other members are). I added `(a.host == nil && b.host == nil)` as a special case for this edge case so the test passes (host should not be null in valid usage)